ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -12,204 +12,142 @@
-    this.state = { value, isValid: true }
-    this.state.isValid = true
-    this.is2DArray = this.is2DArray.bind(this)
-    this.validateListOfPoints = this.validateListOfPoints.bind(this)
-    this.isValidPolygon = this.isValidPolygon.bind(this)
-    this.isValidInput = this.isValidInput.bind(this)
-    this.removeErrorBox = this.removeErrorBox.bind(this)
+function convertWktString(value) {
+  if (value.includes('MULTI')) {
+    return convertMultiWkt(value.includes('POLYGON'), value)
+  } else if (value.includes('POLYGON') && value.endsWith('))')) {
+    return convertWkt(value, 4)
+  } else if (value.includes('LINESTRING') && value.endsWith(')')) {
+    return convertWkt(value, 2)
   }
-  componentWillReceiveProps(props) {
-    if (document.activeElement !== this.ref) {
-      const { geometryKey } = props
-      const value = JSON.stringify(props[geometryKey])
-      this.setState({ value })
-    }
+  return value
+}
+
+function convertWkt(value, numCoords) {
+  const coordinatePairs = value.match(coordinatePairRegex)
+  if (!coordinatePairs || coordinatePairs.length < numCoords) {
+    return value
+  }
+  const coordinates = coordinatePairs.map(coord => coord.replace(' ', ','))
+  return buildWktString(coordinates)
+}
+
+function convertMultiWkt(isPolygon, value) {
+  if (isPolygon && !value.endsWith(')))')) {
+    return value
+  } else if (!value.endsWith('))')) {
+    return value
   }
-  render() {
-    const props = this.props
-    const { label, cursor, geometryKey, unitKey, widthKey } = props
-    return (
-      <React.Fragment>
-        <div className="input-location">
+  const splitter = isPolygon ? '))' : ')'
+  const numPoints = isPolygon ? 4 : 2
+  let shapes = value
+    .split(splitter)
+    .map(shape => shape.match(coordinatePairRegex))
+  shapes = shapes
+    .filter(shape => shape !== null && shape.length >= numPoints)
+    .map(shape => shape.map(coordinatePair => coordinatePair.replace(' ', ',')))
+  return shapes.length === 0
+    ? value
+    : shapes.length === 1
+      ? buildWktString(shapes[0])
+      : '[' + shapes.map(shapeCoords => buildWktString(shapeCoords)) + ']'
+}
+
+const BaseLine = props => {
+  const {
+    label,
+    geometryKey,
+    setState,
+    setBufferState,
+    unitKey,
+    widthKey,
+    mode,
+    polyType,
+  } = props
+  const [currentValue, setCurrentValue] = useState(
+    JSON.stringify(props[geometryKey])
+  )
+  const [baseLineError, setBaseLineError] = useState(initialErrorState)
+  const [bufferError, setBufferError] = useState(initialErrorState)
+
+  useEffect(
+    () => {
+      const { geometryKey } = props
+      setCurrentValue(JSON.stringify(props[geometryKey]))
+      if (props.drawing) {
+        setBaseLineError(initialErrorState)
+        setBufferError(initialErrorState)
+      }
+    },
+    [props.polygon, props.line]
+  )
+
+  return (
+    <div>
+      <div className="input-location">
+        <TextField
+          label={label}
+          value={currentValue}
```

Comment:
see https://github.com/codice/ddf/pull/5814/files#r377803239
```suggestion
          value={JSON.stringify(props[geometryKey])}
```